weave/trace_server/calls_query_builder.py

```diff
@@ -592,17 +605,27 @@ def _as_sql_base_format(
                 c.as_sql(pb, table_alias) for c in self.query_conditions
             )
             for query_condition in self.query_conditions:
-                for field in query_condition._get_consumed_fields():
-                    if isinstance(field, CallsMergedFeedbackPayloadField):
-                        needs_feedback = True
+                if query_condition.is_feedback():
+                    needs_feedback = True
         if self.hardcoded_filter is not None:
             having_conditions_sql.append(self.hardcoded_filter.as_sql(pb, table_alias))
 
+        call_parts_predicate_filters_sql = ""
         if len(having_conditions_sql) > 0:
             having_filter_sql = "HAVING " + combine_conditions(
                 having_conditions_sql, "AND"
             )
 
+        if should_add_predicate_filters(self.query_conditions):
```

Comment:
I would slightly adjust this to be more pluggable.

```
call_parts_predicate_filters_sql = build_call_parts_predicate_filter(self.query_conditions)
``` 

I think that the implementation for "should_add_predicate_filers" is coupled with the implementation of "make_call_parts_predicate_filters_sql" and therefore should be abstracted behind a common method. Looking at your algorithm, i can imagine some alternatives which would make this easier to modify